[Docs] Update model monitoring for v1.7.0 (#5877)
mlrun/mlrun: docs/monitoring/model-monitoring.md

```

The stream function examines 
the log entry, processes it into statistics which are then written to the statistics databases (parquet file, time series database and key value database). 
The monitoring stream function writes the Parquet files using a basic storey ParquetTarget. Additionally, there is a monitoring feature set that refers 
to the same target. You can use `get_offline_features` to read the data from that feature set. 

In parallel, an MLRun job runs, reading the parquet files and performing drift analysis. The drift analysis data is stored so 
that the user can retrieve it in the Iguazio UI or in a Grafana dashboard

When you enable model monitoring, you effectively deploy three components:
- application controller function: handles the monitoring processing and the triggers the apps that trigger the writer. The controller is a scheduled batch job whose frequency is determined by `base_period`. 
- stream function: monitors the log of the data stream. It is triggered when a new log entry is detected. The monitored data is used to create real-time dashboards, detect drift, and analyze performance.
- writer function: writes to the database and outputs alerts.

> 

Reviewer: please remove - redundant

- {py:meth}`~mlrun.projects.MlrunProject.list_model_monitoring_functions` &mdash; Retrieves a list of all the model monitoring functions.
- {py:meth}`~mlrun.projects.MlrunProject.remove_model_monitoring_function` &mdash; Removes the specified model-monitoring-app function from the project and from the DB.
- {py:meth}`~mlrun.projects.MlrunProject.set_model_monitoring_credentials` &mdash; Set the credentials that are used by the project's model monitoring infrastructure functions. 
- {py:meth}`~mlrun.projects.MlrunProject.disable_model_monitoring` &mdash; Disables the controller. 
- {py:meth}`~mlrun.projects.MlrunProject.update_model_monitoring_controller`  &mdash; Redeploys the model monitoring application controller functions.
- {py:meth}`~mlrun.config.Config.get_model_monitoring_file_target_path` &mdash; Gets the full path from the configuration based on the provided project and kind.

## Enable model monitoring

Enable model monitoring for a project with {py:meth}`~mlrun.projects.MlrunProject.enable_model_monitoring`.
The controller runs, by default, every 10 minutes, which is also the minimum interval. 
You can modify the frequency with the parameter `base_period`. 

> To change the `base_period`, call `update_model_monitoring_controller`. 

Reviewer: ```suggestion
To change the `base_period` to 20 minutes, call `update_model_monitoring_controller`. 
```

Author: @assaf758 Are these the only possible values?

Reviewer: no, everything >= 10 minutes is ok.

Author: updated without a specific value

```python

> project.enable_model_monitoring(base_period=1)

Reviewer: ```suggestion
project.enable_model_monitoring(base_period=20)
```

The model monitoring APIs are configured per project. The APIs are:

> - {py:meth}`~mlrun.projects.MlrunProject.enable_model_monitoring` &mdash; Brings up the controller, writer and stream functions, and schedules the controller according to the `base_period`. You can also deploy the default histogram-based data drift application when you enable model monitoring.

Reviewer 2: ```suggestion
- {py:meth}`~mlrun.projects.MlrunProject.enable_model_monitoring` &mdash; Brings up the controller, writer and stream real time functions, and schedules the controller according to the `base_period`. You can also deploy the default histogram-based data drift application when you enable model monitoring.
```

The controller checks for new datasets every `base_period` to send to the app. Invoking the model a second time ensures that the previous 
window closed and therefore the data contains the full monitoring window. From this point on, the applications are triggered by the controller. 

> The controller checks the Parquet DB every 10 minutes (or non-default 

Reviewer: ```suggestion
The controller checks the Parquet DB every 10 minutes (or higher number, configurable by the user)
```
